File: best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    n := len(prices)
    minprice := prices[0]
    maxprofit := 0
    for i:=1; i<n; i++{
        if prices[i] < minprice{
            minprice = prices[i]
        }
        if maxprofit < prices[i] - minprice{
            maxprofit = prices[i] - minprice
        }
    }
    return maxprofit
}

File: best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        minprices = prices[0]
        maxprofit = 0
        
        for i in range(len(prices)):
            if minprices > prices[i]:
                minprices = prices[i]
            if maxprofit < prices[i] - minprices:
                maxprofit = prices[i] - minprices
        return maxprofit